test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py

```diff
@@ -196,6 +204,153 @@ def test_fsdp_ddp_parity_with_grad_scaler(
                 sharded_grad_scaler_kwargs=sharded_grad_scaler_kwargs,
             )
 
+    def _build_model_and_optim(
+        self,
+        build_extra_ddp: bool = False,
+        cpu_offload: CPUOffload = CPUOffload(offload_params=False),
+        use_orig_params: bool = False,
+    ):
+        model = TransformerWithSharedParams.init(
+            self.process_group,
+            FSDPInitMode.NO_FSDP,
+            CUDAInitMode.CUDA_BEFORE,
+            deterministic=True,
+        )
+
+        if build_extra_ddp:
+            ref_model = DDP(
+                copy.deepcopy(model),
+                device_ids=[self.rank],
+            )
+            ref_optim = torch.optim.SGD(ref_model.parameters(), lr=1e-2)
```

Comment:
nit (for future reference): It could be good to use `Adam` since an Adam-like optimizer is most commonly used with FSDP.